### PR TITLE
[Batch] `az batch`: Fix batch cloud console authentication issue

### PR DESCRIPTION
**Pull request by wanghoppe**
**Related command**
az batch related commands

**Description**
fix Batch cloud console authentication issue, use batch_resource_id to request AAD tokens to authenticate with Batch Dataplane APIs.

**Testing Guide**
N/A

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by dpwatrous** [comment]
Looks good to me

**Comment by wanghoppe** [comment]
Hi @zhoxing-ms, could you help to take a look at this PR? Thanks.

File: src/azure-cli/azure/cli/command_modules/batch/_client_factory.py [hunk 1]
```diff
@@ -89,12 +87,7 @@ def batch_data_service_factory(cli_ctx, kwargs):
     if not account_key:
         from azure.cli.core._profile import Profile
         profile = Profile(cli_ctx=cli_ctx)
-        # in order to use AAD auth in cloud shell mode, we will use mgmt AAD token
-        # instead of Batch AAD token to auth
-        if in_cloud_console():
-            resource = cli_ctx.cloud.endpoints.active_directory_resource_id
-        else:
-            resource = cli_ctx.cloud.endpoints.batch_resource_id
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/batch/_client_factory.py:97]: May I ask why has this logic changed?

Comment (wanghoppe) [reply src/azure-cli/azure/cli/command_modules/batch/_client_factory.py:97]: The logic was added in this PR: https://github.com/Azure/azure-cli/pull/6582, I asked the PR owner, and he seems to forget the details. Maybe it's been broken for a long time and no one spotted it. Using the management endpoint token will result in customer isn't able to authenticate with our data plane, which generally only accepts Batch endpoint resource tokens. I already tested it in the cloud shell environment and verified this fix will resolve the customer's error.

**Comment by wanghoppe** [comment]
Hi @zhoxing-ms, please let me know if you have any questions.

File: src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_mgmt_commands.py [hunk 2]
```diff
@@ -54,7 +54,8 @@ def test_batch_general_arm_cmd(self, resource_group, storage_account):
             self.check('encryption.keySource', 'Microsoft.Batch'),
             self.check('resourceGroup', '{rg}')])
 
-        time.sleep(100)
+        if self.is_live or self.in_recording:
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_mgmt_commands.py:57]: Why do we need to add this logic?

Comment (wanghoppe) [reply src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_mgmt_commands.py:57]: To speed up the test in playback mode by only `sleep` in live or recording mode. It's a common pattern in our test code, I just find we forget to wrap it here so adding it here.

**Comment by wanghoppe** [comment]
Hi @zhoxing-ms, do you have any other questions for this PR?

File: src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_mgmt_commands.py (hunk 2 above)
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_mgmt_commands.py:57]: Actually, the logic of `self.in_recording` already includes the logic of `self.is_live`.

**Comment by wanghoppe** [comment]
Hi @zhoxing-ms, I push a new commit to remove `self.in_recording`. This PR fixes a production issue that customers can't use `az batch` to communicate with our data plane on Cloudshell. We wish to have it merged and released as soon as possible. Let me know if you have other questions.

File: src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_mgmt_commands.py (hunk 2 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_mgmt_commands.py]: @wanghoppe Actually, please note that the logic of `self.in_recording` containing `self.is_live`, so it should be using `self.in_recording` ?

Comment (wanghoppe) [reply src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_mgmt_commands.py]: @zhoxing-ms Thanks for reminding, changed it to `self.in_recording` instead

**Comment by wanghoppe** [comment]
Hi @zhoxing-ms gentle ping for reviewing

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**